frontend/src/ui/Access.tsx

```diff
@@ -136,230 +111,219 @@ type AclSelectProps = SelectProps & {
-                if (b.value === getUserRole(user)) {
-                    return 1;
-                }
-                // Otherwise show entries in order of addition.
-                return 0;
+            // A is a subset of b, so b should come first.
+            if (isSubset(a.value, b.value)) {
+                return 1;
             }
-        };
+            // B is a subset of a, so a should come first.
+            if (isSubset(b.value, a.value)) {
+                return -1;
+            }
+            // Neither is a subset of the other, don't sort.
+            return 0;
+        } else {
+            // Always show the current user first, if included.
+            if (a.value === getUserRole(user)) {
+                return -1;
+            }
+            if (b.value === getUserRole(user)) {
+                return 1;
+            }
+            // Otherwise show entries in order of addition.
+            return 0;
+        }
+    };
 
-        const initialSelections: Option[] = makeSelection(
-            kind === "Group" ? DUMMY_GROUPS : DUMMY_USERS, initialAcl
-        ).sort((roleComparator));
+    const initialSelections: Option[] = makeSelection(
+        kind === "Group" ? DUMMY_GROUPS : DUMMY_USERS, initialAcl
+    ).sort((roleComparator));
 
-        const initialOptions = allOptions.filter(
-            item => !initialSelections.some(elem => elem.value === item.value)
-        );
+    const initialOptions = allOptions.filter(
+        item => !initialSelections.some(elem => elem.value === item.value)
+    );
 
-        // The ACL might not explicitly include admin, but since we still want to show
-        // the admin entry when logged in as admin, the item needs to be added manually.
-        if (kind === "User" && !initialSelections.some(
-            selection => selection.label === "Administrator"
-        )) {
-            initialSelections.splice(0, 0, defaultDummyOption);
-        }
+    // The ACL might not explicitly include admin, but since we still want to show
+    // the admin entry when logged in as admin, the item needs to be added manually.
+    if (kind === "User" && !initialSelections.some(
+        selection => selection.label === "Administrator"
+    )) {
+        initialSelections.splice(0, 0, defaultDummyOption);
+    }
 
-        const [selection, setSelection] = useState<MultiValue<Option>>(initialSelections);
-        const [options, setOptions] = useState<MultiValue<Option>>(initialOptions);
+    const [selection, setSelection] = useState<MultiValue<Option>>(initialSelections);
+    const [options, setOptions] = useState<MultiValue<Option>>(initialOptions);
 
-        useImperativeHandle(ref, () => ({
-            getSelection: () => selection,
-            reset: () => {
-                setSelection(initialSelections);
-                setOptions(initialOptions);
-            },
-        }));
+    useEffect(() => {
+        setSelection(initialSelections);
+        setOptions(initialOptions);
```

Comment:
Why is that necessary :thinking: I feel like we shouldn't need an effect here.

I think:
- The prop `initialAcl` should be renamed somehow (`acl`, but that clashes with something else...). Because it's not initial. It's always the current ACL that's displayed.
- Further, I think you can completely remove the two `useState<MultiValue<...>>` calls and just rename `initialSelections` to `selections` and same with `initialOptions`. Again, those are directly derived from the acls coming in. And whenever those change, they are changed as well.
- And then you should be able to remove this `useEffect`
- And then you can also remove all `setSelection` and `setOptions` calls from the handlers below. That should just work I think.

Reply:
>Further, I think you can completely remove the two useState<MultiValue<...>>

I was under the impression that those are needed for react select, but I guess if you say it can be done without them, then it can be done without them.